aspnet/fundamentals/logging.rst

```diff
@@ -126,105 +180,251 @@ Debug          dbug
+  dbug: Microsoft.AspNetCore.Mvc.Internal.ActionSelector[2]
+        Action 'TodoApi.Controllers.TodoController.Delete (TodoApi)' with id '529c0e82-aea6-466c-bbe2-e77ded858853' did not match the constraint 'Microsoft.AspNetCore.Mvc.Internal.HttpMethodActionConstraint'
+  dbug: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[1]
+        Executing action TodoApi.Controllers.TodoController.GetById (TodoApi)
+  info: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[1]
+        Executing action method TodoApi.Controllers.TodoController.GetById (TodoApi) with arguments (0) - ModelState is Valid
+  info: TodoApi.Controllers.TodoController[1002]
+        Getting item 0
+  warn: TodoApi.Controllers.TodoController[4000]
+        GetById(0) NOT FOUND
+  dbug: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[2]
+        Executed action method TodoApi.Controllers.TodoController.GetById (TodoApi), returned result Microsoft.AspNetCore.Mvc.NotFoundResult.
+  info: Microsoft.AspNetCore.Mvc.StatusCodeResult[1]
+        Executing HttpStatusCodeResult, setting HTTP status code 404
+  info: Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker[2]
+        Executed action TodoApi.Controllers.TodoController.GetById (TodoApi) in 198.8402ms
+  info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
+        Request finished in 550.6837ms 404
+  dbug: Microsoft.AspNetCore.Server.Kestrel[9]
+        Connection id "0HKV8M5ARIH5P" completed keep alive response.
+
+Log scopes
+----------
+
+You can group a set of logical operations within a *scope* in order to attach the same data to each log that is created as part of that set.  For example, you might want every log created as part of processing a transaction to include the transaction ID.
+
+Scopes are not required, are not supported by all providers, and should be used sparingly. They're best used for operations that have a distinct beginning and end, such as a transaction involving multiple resources.
+
+A scope is an ``IDisposable`` type returned by calling the ``ILogger.BeginScope<TState>`` method, which lasts from the moment it is created until it is disposed. You use a scope by wrapping your logger calls in a ``using`` block, as shown here:
 
 .. literalinclude:: logging/sample/src/TodoApi/Controllers/TodoController.cs
   :language: c#
-  :lines: 32-43
-  :emphasize-lines: 8
+  :start-after: snippet_Scopes
+  :end-before: snippet_Scopes
+  :emphasize-lines: 4-5,13
   :dedent: 8
 
-To test out this code, you can trigger logging a warning by running the app from the console and navigating to ``http://localhost:5000/api/Todo/0``. You should see output similar to the following:
+The Console logging provider supports scopes. To use them, you pass in an ``includeScopes`` parameter to the ``AddConsole`` extension method, as shown here:
```

Comment:
The sample below uses enables scopes on the console logger:

I don't think you want to Capitalize Console.  Don't verbalize the code. Explain it when necessary, but don't repeat in English.